### PR TITLE
[Android] Add integrity/signature check in download mode

### PR DESCRIPTION
**Pull request by XiaosongWei**
Signature check is enabled by default and it requires that the crosswalk runtime APK must be signed with the same key used in application APK signing. Developer can disable the signature check by inserting a meta-data in AndroidManifest.

Related to XWALK-5121

### CODE REVIEW

**Comment by XiaosongWei** [comment]
@lincsoon @rakuco @baleboy  @halton Please take a look, Thanks!
@terriko Please help to review from security point of view, Thanks!

File: runtime/android/core/src/org/xwalk/core/XWalkUpdater.java [hunk 1]
```diff
@@ -216,6 +217,11 @@
     private static final String XWALK_CORE_EXTRACTED_DIR = "extracted_xwalkcore";
     private static final String ARCH_QUERY_STRING = "?arch=";
 
+    // Hash for Chromium debug key
+    private static final String XWALK_APK_HASH_ALGORITHM = "SHA-256";
+    private static final String XWALK_APK_HASH_CODE =
+            "32a2fc74d731105859e5a85df16d95f102d85b22099b8064c5d8915c61dad1e0";
```
Comment (lincsoon) [inline runtime/android/core/src/org/xwalk/core/XWalkUpdater.java]: Please move these signature-related constants to XWalkAppVersion.template

**Comment by lincsoon** [comment]
@baleboy The content of `XWalkCoreVerifier` comes from the shared mode and has already been reviewed by the security team before. Please help to contact some one in security team to review the part about silent download mode.

**Comment by XiaosongWei** [comment]
@baleboy @terriko New signature check implementation: now it requires that the crosswalk runtime APK must be signed with the same key used in application APK signing. Please take a look, Thanks!

**Comment by terriko** [comment]
Are you checking the signatures or the hashes of the apk?  The file should have both a signature and a set of hashes for each file and it looks like you're checking hashes but not the signature.  (It may not matter for this case, I just want to be clear.)

We may be able to bytecompare things directly rather than doing the digest, but I'm not sure if that's a performance improvement or not, although it would avoid hash collisions.

**Comment by XiaosongWei** [comment]
@terriko It's checking the signatures of the APKs. Yes, it's possible to do bytecompare directly and signature.equals() is doing bytecompare internally. 

```
appPackageInfo.signatures[i].equals(runtimePackageInfo.signatures[i])
```

Now I changed it to use bytecompare, please take a look again. Thanks!

**Comment by terriko** [comment]
lgtm!

**Comment by XiaosongWei** [comment]
@lincsoon @halton Please help to merge if no more comments, thanks!

**Pull request merged**